kernel/src/loader/mod.rs

```diff
@@ -144,6 +148,76 @@ where
     Ok(GuestAddress(ehdr.e_entry as usize))
 }
 
+#[cfg(target_arch = "aarch64")]
+pub fn load_kernel<F>(
+    guest_mem: &GuestMemory,
+    kernel_image: &mut F,
+    start_address: usize,
+) -> Result<GuestAddress>
+where
+    F: Read + Seek,
+{
+    /* Kernel boot protocol is specified in the kernel docs
+     * Documentation/arm/Booting and Documentation/arm64/booting.txt
+     */
+    const AARCH64_KERNEL_LOAD_ADDR: usize = 0x80000;
+    const AARCH64_MAGIC_NUMBER: u32 = 0x644d5241;
+    let mut kernel_load_offset = AARCH64_KERNEL_LOAD_ADDR;
+
+    /* Look for the magic number inside the elf header. */
+    let magic_offset_header = 2 * mem::size_of::<u32>() + 6 * mem::size_of::<u64>(); // This should total 56.
+
+    kernel_image
+        .seek(SeekFrom::Start(magic_offset_header as u64))
+        .map_err(|_| Error::SeekKernelImage)?;
+    let mut magic_number: u32 = 0;
+    unsafe {
+        sys_util::read_struct(kernel_image, &mut magic_number)
+            .map_err(|_| Error::ReadProgramHeader)?
+    }
+    if magic_number != AARCH64_MAGIC_NUMBER {
+        return Err(Error::InvalidElfMagicNumber);
+    }
+
+    /* Look for the `text_offset` from the elf header. */
+    kernel_image
+        .seek(SeekFrom::Start(0))
+        .map_err(|_| Error::SeekKernelImage)?;
+    kernel_image
+        .seek(SeekFrom::Start(2 * mem::size_of::<u32>() as u64)) // This should total 8.
```

Comment:
nit: I would add a constant for `text_offset = 2 * mem::size_of::<u32>() as u64` for code readability.